code/modules/mob/living/silicon/robot/emote.dm

```diff
@@ -207,17 +224,31 @@
 			else
 				to_chat(src, "You are not security.")
 
-		/*
-		if ("fart")
-			var/list/robotfarts = list("makes a farting noise","vents excess methane","shakes violently, then vents methane.")
-			var/robofart = pick(robotfarts)
-			message = "<B>[src]</B> [robofart]."
-			m_type = VISIBLE
-
-		*/
+		if ("vent")
+			if(world.time-lastvent >= 30 SECONDS)
+				var/list/robotfarts = list("vents","shakes violently, then vents")
+				var/robofart = pick(robotfarts)
+				for(var/mob/living/M in view(0))
+					if(M != src && M.loc == src.loc)
+						if(istype(M, /mob/living/carbon/human/vox))
+							visible_message("<span class = 'warning'><b>[src]</b> [robofart] oxygen in <b>[M]</b>'s face!</span>")
+							if (M.internal != null && M.wear_mask && (M.wear_mask.clothing_flags & MASKINTERNALS))
+								continue
+							if(M.reagents)
+								M.reagents.add_reagent(OXYGEN,rand(1,15))
+								src.attack_log += text("\[[time_stamp()]\] <font color='orange'>[src] vented oxygen in [M]'s face.</font>")
```

Comment:
There's a proc for adding to attack logs, see https://github.com/d3athrow/vgstation13/pull/12949/files